File: tools/file.js
const fs = require('fs');
const readline = require('readline');

methods = {
  readLinesTo2DArray: function(filePath, callBack) {
    var inputs = [];
    const rl = readline.createInterface({
      input: fs.createReadStream(filePath),
      crlfDelay: Infinity
    });
    rl.on('line', (line) => {
        var lineArray = [];
        var strings = line.split(' ');
        for (var i = 0; i < strings.length; i++) {
            lineArray.push(Number(strings[i]));
        }
      inputs.push(lineArray);
    }).on('close', () => {
        callBack(inputs);
    });
  }
}

module.exports = methods;

File: tools/file.py
# File format expected:
# 1 2 3
# 4 5 6
# 7 8 9

cellDelimiter = ' '

def readLinesTo2DArray(filePath):
    array2D = []
    with open(filePath, 'r') as openfile:
        i = 0
        for line in openfile:
            array2D.append([])
            array2D[i] = [int(n) for n in line.split(cellDelimiter)]
            i += 1
    return array2D


